github_issue_summarization/notebooks/Makefile

```diff
@@ -70,38 +70,39 @@ build:
 	@echo Built $(IMG):$(TAG)
 
 $(BUILD_DIR)/body_pp.dpkl:
-	mkdir -p model
+	mkdir -p $(BUILD_DIR)
 	gsutil cp $(MODEL_GCS)/body_pp.dpkl $(BUILD_DIR)/
 
 $(BUILD_DIR)/title_pp.dpkl:
-	mkdir -p model
+	mkdir -p $(BUILD_DIR)
 	gsutil cp $(MODEL_GCS)/title_pp.dpkl $(BUILD_DIR)/
 
 $(BUILD_DIR)/seq2seq_model_tutorial.h5:
-	mkdir -p model
+	mkdir -p $(BUILD_DIR)
 	gsutil cp $(MODEL_GCS)/seq2seq_model_tutorial.h5 $(BUILD_DIR)/
 
 # Copy python files into the model directory
 # so that we can mount a single directory into the container
 $(BUILD_DIR)/% : %
+	mkdir -p $(BUILD_DIR)
 	cp -f $< $@
 
 download-model:	$(BUILD_DIR)/seq2seq_model_tutorial.h5 $(BUILD_DIR)/title_pp.dpkl $(BUILD_DIR)/body_pp.dpkl
 
-# TODO(jlewi): This doesn't actually pick up changes to code. The problem is that docker run is not detecting when code changes
-# and when it does copying it to $(BUILD_DIR)/build. Using --force fixes that problem but then it looks like docker build
-# no longer uses the cache and rebuilds are slow. Manually copying the files doesn't work because the files end up
-# owned by root because they were created inside the container.
-# The work around now is to manually copy files (e.g. in a  shell outside Make) as needed
-build-model-image: download-model $(BUILD_DIR)/seq2seq_utils.py $(BUILD_DIR)/IssueSummarization.py $(BUILD_DIR)/requirements.txt
-	# The docker run comand creates a Dockerfile for the seldon image with required assets
-	docker run -v $(BUILD_DIR):/my_model seldonio/core-python-wrapper:0.7 /my_model IssueSummarization $(TAG) gcr.io --base-image=python:3.6 --image-name=$(MODEL_IMG_NAME)
-	# We don't use the script generated by Seldon because that script won't get updated by make to reflect the change
-	# in the desired tag.
-	docker build -t $(MODEL_IMG):$(TAG) -f $(BUILD_DIR)/build/Dockerfile $(BUILD_DIR)/build
+build-model-image: download-model $(BUILD_DIR)/seq2seq_utils.py $(BUILD_DIR)/IssueSummarization.py $(BUILD_DIR)/requirements.txt $(BUILD_DIR)/environment_seldon_rest
+	cd $(BUILD_DIR) && s2i build -E environment_seldon_rest . seldonio/seldon-core-s2i-python36:0.4 $(MODEL_IMG):$(TAG)
 	@echo built $(MODEL_IMG):$(TAG)
 
-push-model-image: build-model-image
+start-docker-model-image:
```

Comment:
no action needed; just an FYI

My current thinking is that a better way to begin testing is by writing a manual test; e.g. a python file that can send predictions and validate the results; something like
https://github.com/kubeflow/examples/blob/master/mnist/testing/deploy_test.py

To begin with this can be run as a manual test where the model can be running either locally (via docker) or on a cluster. We can use kubectl port-forward to setup a local port if needed to use.